src/librustc/ty/maps.rs

```diff
@@ -804,6 +845,10 @@ fn metadata_dep_node(def_id: DefId) -> DepNode<DefId> {
     DepNode::MetaData(def_id)
 }
 
+fn cratenum_metadata_dep_node(crate_num: CrateNum) -> DepNode<DefId> {
+    DepNode::MetaDataByCrateNum(crate_num)
```

Comment:
I don't think we should have `DepNode` with a cratenum in it, though, at least not yet, as it won't get remapped properly when being loaded back up.